examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,361 @@
+    }
+
+    return new Stack(
+      alignment: Alignment.bottomCenter,
+      fit: StackFit.passthrough,
+      children: children,
+    );
+  }
+}
+
+class FadeAnimation extends StatefulWidget {
+  final Widget child;
+  final Duration duration;
+
+  const FadeAnimation(
+      {this.child, this.duration: const Duration(milliseconds: 500)});
+
+  @override
+  _FadeAnimationState createState() => new _FadeAnimationState();
+}
+
+class _FadeAnimationState extends State<FadeAnimation>
+    with SingleTickerProviderStateMixin {
+  AnimationController animationController;
+
+  @override
+  void initState() {
+    super.initState();
+    animationController =
+        new AnimationController(duration: widget.duration, vsync: this);
+    animationController.addListener(() {
+      if (mounted) {
+        setState(() {});
+      }
+    });
+    animationController.forward(from: 0.0);
+  }
+
+  @override
+  void deactivate() {
+    animationController.stop();
+    super.deactivate();
+  }
+
+  @override
+  void didUpdateWidget(FadeAnimation oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.child != widget.child) {
+      animationController.forward(from: 0.0);
+    }
+  }
+
+  @override
+  void dispose() {
+    animationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return animationController.isAnimating
+        ? new Opacity(
+            opacity: 1.0 - animationController.value,
+            child: widget.child,
+          )
+        : new Container();
+  }
+}
+
+// Shows a
+class ConnectivityOverlay extends StatefulWidget {
+  final Widget child;
+  final Completer<Null> connectedCompleter;
+  final GlobalKey<ScaffoldState> scaffoldKey;
+
+  const ConnectivityOverlay(
+      {this.child, this.connectedCompleter, this.scaffoldKey});
+
+  @override
+  _ConnectivityOverlayState createState() => new _ConnectivityOverlayState();
+}
+
+class _ConnectivityOverlayState extends State<ConnectivityOverlay> {
+  StreamSubscription<ConnectivityResult> connectivitySubscription;
+  bool connected = true;
+
+  static const Widget errorSnackBar = const SnackBar(
+    backgroundColor: Colors.red,
+    content: const ListTile(
+      title: const Text('No network'),
+      subtitle: const Text(
+          'To load the videos you must have an active network connection'),
+    ),
+  );
+
+  Stream<ConnectivityResult> connectivityStream() async* {
+    final Connectivity connectivity = new Connectivity();
+    ConnectivityResult previousResult = await connectivity.checkConnectivity();
+    yield previousResult;
+    await for (ConnectivityResult result
+        in connectivity.onConnectivityChanged) {
+      if (result != previousResult) {
+        yield result;
+        previousResult = result;
+      }
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    connectivitySubscription =
+        connectivityStream().listen((ConnectivityResult connectivityResult) {
```

Comment:
Dunno if this might help:

```dart
connectivitySubscription = connectivityStream().listen(
  (ConnectivityResult connectivityResult) {
    ...
  },
);
```

Reply:
done